### PR TITLE
[Chore] Audit Style Guide Headings

### LINKED ISSUE #337: [Chore] Audit Style Guide Headings
### What general chore or task would you like to see performed?
With #310, we need to audit all headings in the style guide to ensure that the proper sizes are used.

### Are there any specific steps you would like completed?
1. Match all heading classes to the corresponding html element (`<h3 class="go-heading-3"></h3>`)
2. Change the headings to the proper size per the new sizes

### PR DESCRIPTION
**Pull request by grahamhency**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Most of our `<h*>` tags did not match the new `go-heading-x` classes provided from UX.

Issue Number: Closes #337 

## What is the new behavior?
This updates every heading in the style guide to match correctly.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-overview/accordion-overview.component.html [hunk 1]
```diff
@@ -1,106 +1,116 @@
...
   <go-card class="go-column go-column--100">
     <header go-card-header>
-      <h3 class="go-heading-3">Default</h3>
+      <h2 class="go-heading-2">Default</h2>
     </header>
     <div go-card-content>
       <div class="go-container">
         <div class="go-column go-column--50">
-          <h4 class="go-heading-4">Code</h4>
+          <h3 class="go-heading-3">Code</h3>
```
Comment (StevenUlmer) [inline projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-overview/accordion-overview.component.html]: In forms we're using an h4 with a class of `go-heading--underlined` for the code and view headers. Should we make these consistent?

File: projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html [hunk 2]
```diff
@@ -1,66 +1,75 @@
...
   <go-card class="go-column go-column--100">
     <header go-card-header>
-      <h1 class="go-heading-5">Default</h1>
+      <h2 class="go-heading-2">Default</h2>
     </header>
     <div go-card-content>
       <div class="go-container">
         <div class="go-column go-column--50">
-          <h2 class="go-heading-6">Code</h2>
+          <h3 class="go-heading-3">Code</h3>
```
Comment (StevenUlmer) [inline projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html]: should these be an h4 with underlined?

File: projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-overview/action-sheet-overview.component.html [hunk 3]
```diff
@@ -51,11 +54,11 @@ <h1 class="go-heading-5">Simple</h1>
           </p>
         </div>
         <div class="go-column go-column--50 go-column--no-padding">
-          <h2 class="go-heading-6">Code</h2>
+          <h3 class="go-heading-3">Code</h3>
```
Comment (StevenUlmer) [inline projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-overview/action-sheet-overview.component.html]: should these be an h4 with underlined?

File: projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-panel-docs/action-sheet-panel-docs.component.html [hunk 4]
```diff
@@ -59,11 +72,11 @@ <h1 class="go-heading-5">Icons</h1>
           </p>
         </div>
         <div class="go-column go-column--50 go-column--no-padding">
-          <h2 class="go-heading-6">Code</h2>
+          <h3 class="go-heading-3">Code</h3>
```
Comment (StevenUlmer) [inline projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-panel-docs/action-sheet-panel-docs.component.html]: should these be an h4 with underlined?

File: projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html [hunk 5]
```diff
@@ -1,66 +1,75 @@
-<header class="go-page-title">
-  <h1 class="go-heading-3">{{ pageTitle }}</h1>
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html]: @StevenUlmer yeah, this uses the `subNavService` to render a title so this is unnecessary.

File: projects/go-style-guide/src/app/features/ui-kit/components/badge-docs/badge-docs.component.html [hunk 6]
```diff
@@ -1,120 +1,128 @@
...
+          Setting this to false will cause the badge to just be an empty circle without displaying
+          the `badgeData`.
+        </p>
+      </div>
+
+    </div>
+  </go-card>
+
+  <go-card class="go-column go-column--100">
+    <ng-container go-card-header>
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/badge-docs/badge-docs.component.html]: I don't know if I agree with this exactly. If you need to use a `header` then use a `header`.

File: projects/go-style-guide/src/app/features/ui-kit/components/badge-docs/badge-docs.component.html [hunk 7]
```diff
@@ -1,120 +1,128 @@
...
+      <h2 class="go-heading-2">Basic Example</h2>
+    </ng-container>
+    <ng-container go-card-content>
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/badge-docs/badge-docs.component.html]: 🤷‍♂

File: projects/go-style-guide/src/app/features/ui-kit/components/icon-button-docs/icon-button-docs.component.html [hunk 8]
```diff
@@ -1,146 +1,149 @@
...
-        <div class="go-column go-column--50">
-          <code [highlight]="disabledExample"></code>
-        </div>
-        <div class="go-column go-column--50">
-          <go-icon-button (handleClick)="testClick()" buttonIcon="home" buttonDisabled="true"></go-icon-button>
-        </div>
-      </section>
+      <h2 class="go-heading-2">Button Disabled</h2>
     </ng-container>
+    <section class="go-container" go-card-content>
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/icon-button-docs/icon-button-docs.component.html]: 🤷‍♂

File: projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/text-area-docs/text-area-docs.component.html [hunk 9]
```diff
@@ -1,249 +1,303 @@
...
+      </div>
+
+      <div class="go-column go-column--100 go-column--no-padding">
+        <div class="go-container">
+          <div class="go-column go-column--50">
+            <h4 class="go-heading-4 go-heading--underlined">View</h4>
+            <go-text-area [control]="message"></go-text-area>
+          </div>
+          <div class="go-column go-column--50">
+            <h4 class="go-heading-64go-heading--underlined">Code</h4>
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/form-docs/components/text-area-docs/text-area-docs.component.html]: Good catch

File: projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-overview/accordion-overview.component.html (hunk 1 above)
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-overview/accordion-overview.component.html]: Sure, I fixed it.

File: projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html (hunk 2 above)
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.html]: Sure, I fixed it.

File: projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-overview/action-sheet-overview.component.html (hunk 3 above)
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-overview/action-sheet-overview.component.html]: Sure, I fixed it.

File: projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-panel-docs/action-sheet-panel-docs.component.html (hunk 4 above)
Comment (grahamhency) [reply projects/go-style-guide/src/app/features/ui-kit/components/action-sheet-docs/components/action-sheet-panel-docs/action-sheet-panel-docs.component.html]: Sure, I fixed it.

**Pull request merged**